docs/src/tutorials/temporal-properties.md

```diff
@@ -0,0 +1,416 @@
+    /\ __loop_InLoop = FALSE
+    ...
+    /\ __loop_isGreen = FALSE
+    /\ __loop_requestedGreen = FALSE
+    ...
+    /\ isGreen = FALSE
+    /\ requestedGreen = TRUE *)
+
+(* State3 ==
+    ...
+    /\ __loop_InLoop = TRUE
+    ...
+    /\ __loop_isGreen = FALSE
+    /\ __loop_requestedGreen = TRUE
+    ...
+    /\ isGreen = FALSE
+    /\ requestedGreen = TRUE *)
+```
+So, initially, `isGreen` and `requestedGreen` are both false.
+Further, `__loop_InLoop` is false, and the copies of `isGreen` and `requestedGreen`, which are called
+`__loop_isGreen` and `__loop_requestedGreen`, are equal to the values of `isGreen` and `requestedGreen`.
+
+From state 0 to state 1, `requestedGreen` changes from false to true.
+From state 1 to state 2, the system stutters, and the valuation of model variables remains unchanged.
+Finally, in state 3 `__loop_InLoop` is set to true, which means that
+the loop starts in state 2, and the trace from state 3 onward is inside the loop.
+However, since state 3 is the last state, this means simply that the trace loops in state 2.
+Since the loop starts, the copies of the system variables are also set to the values of the variables in state 2,
+so ` __loop_isGreen = FALSE` and `__loop_requestedGreen = TRUE`.
+
+
+The looping trace in this case can be visualized like this:
+
+![A counterexample trace for the property [](requestedGreen => <>isGreen)](img/counterexample.png)
+
+It is also clear why this trace violates the property:
+`requestedGreen` holds in state 1, but `isGreen` never holds,
+so in state 1 the property `requestedGreen => <>isGreen` is violated.
+
+## Auxiliary variables encode evaluations of subformulas along the trace
+
+Next, let us discuss the other auxiliary variables that are introduced by Apalache to check the temporal property.
+These extra variables correspond to parts of the temporal property we want to check.
+These are the following variables with their valuations in the initial state:
+
+```
+(* State0 ==
+  RequestWillBeFulfilled_init = FALSE
+    ...
+    /\ ☐(requestedGreen ⇒ ♢isGreen) = FALSE
+    /\ ☐(requestedGreen ⇒ ♢isGreen)_unroll = TRUE
+    /\ requestedGreen ⇒ ♢isGreen = TRUE
+    /\ ♢isGreen = FALSE
+    /\ ♢isGreen_unroll = FALSE
+    ...
+```
+There are three groups of variables: 
+* Variables that look like formulas, e.g. `☐(requestedGreen ⇒ ♢isGreen)`
+* Variables that look like formulas and end with `_unroll`,
+e.g. `☐(requestedGreen ⇒ ♢isGreen)_unroll`
+* The variable `RequestWillBeFulfilled_init`.
+
+Let's focus on the non-`_unroll` variables that look like formulas
+first.
+
+Recall that the temporal property we want to check is `[](requestedGreen => <>isGreen)`.
+That's also the name of one of the variables: The value of the variable 
+`☐(requestedGreen ⇒ ♢isGreen)` tells us whether starting in the current state, the
+formula `[](requestedGreen => <>isGreen)` holds. Since we are looking at a counterexample, it is not
+surprising that the formula does not hold in state 0.
```

Comment:
```suggestion
formula `[](requestedGreen => <>isGreen)` holds. Since we are looking at a counterexample to this formula, it should not be
surprising that the formula does not hold in the initial state of the counterexample.
```